### PR TITLE
Child Hotkey refactor

### PR DESCRIPTION
**Pull request by opendansor**
Child hotkey refactor

### CODE REVIEW

File: bittensor/commands/stake.py [hunk 1]
```diff
@@ -660,13 +662,15 @@ def _run(cli: "bittensor.cli", subtensor: "bittensor.subtensor"):
             if not wallet_utils.is_valid_ss58_address(child):
                 console.print(f":cross_mark:[red] Invalid SS58 address: {child}[/red]")
                 return
-
+        console.print(f"the number of children is {len(children)}")
         if (
             len(children) == 1
         ):  # if only one child, then they have full proportion by default
             cli.config.proportions = 1.0
```
Comment (galina-opentensor) [inline bittensor/commands/stake.py]: The proportions parameter is a float, not a string. Please take a look.
```suggestion
            cli.config.proportions = "1.0"
```

File: bittensor/extrinsics/staking.py [hunk 2]
```diff
@@ -567,17 +568,99 @@ def set_children_extrinsic(
...
 
-    user_hotkey_ss58 = wallet.hotkey.ss58_address  # Default to wallet's own hotkey.
-    if hotkey != user_hotkey_ss58:
-        raise ValueError("Can only call  children for other hotkeys.")
+    with bittensor.__console__.status(
+        f":satellite: Setting childkey take on [white]{subtensor.network}[/white] ..."
+    ):
+        try:
+            
+            if 0 < take < 0.18:
```
Comment (galina-opentensor) [inline bittensor/extrinsics/staking.py]:
```suggestion
            if 0 < take <= 0.18:
```

File: bittensor/commands/stake.py [hunk 3]
```diff
@@ -36,11 +36,43 @@
 )
 from . import defaults  # type: ignore
 from ..utils import wallet_utils
-from ..utils.formatting import u64_to_float
+from ..utils.formatting import u64_to_float, u16_to_float
 
 console = bittensor.__console__
 
 
+def get_netuid(cli, subtensor):
```
Comment (basfroman) [inline bittensor/commands/stake.py]: pls add single line docstring

File: bittensor/commands/stake.py [hunk 4]
```diff
@@ -36,11 +36,43 @@
...
+        console.print(
+            "[red]Network with netuid {} does not exist. Please try again.[/red]".format(
+                netuid
+            )
+        )
+        return False, -1
+    return True, netuid
+
+
+def get_hotkey(wallet: "bittensor.wallet", config: "bittensor.config") -> str:
```
Comment (basfroman) [inline bittensor/commands/stake.py:70]: pls add single line docstring

File: bittensor/commands/stake.py [hunk 5]
```diff
@@ -610,81 +895,94 @@ def run(cli: "bittensor.cli"):
...
         if not cli.config.is_set("children"):
             cli.config.children = Prompt.ask(
-                "Enter child(ren) hotkeys (ss58) as comma-separated values"
+                "Enter child hotkeys (ss58) as comma-separated values"
             )
-        children = [str(x) for x in re.split(r"[ ,]+", cli.config.children)]
+        proposed_children = [str(x) for x in re.split(r"[ ,]+", cli.config.children)]
+
+        # Set max 5 children
+        if len(proposed_children) > 5:
```
Comment (basfroman) [inline bittensor/commands/stake.py]: maybe it's worth keeping this (`5`) as a constant at the top of the module to be more obvious about this limit?

File: bittensor/commands/stake.py [hunk 6]
```diff
@@ -573,6 +605,259 @@ def add_args(parser: argparse.ArgumentParser):
...
+            )
+            GetChildKeyTakeCommand._run(cli, subtensor)
+        finally:
+            if "subtensor" in locals():
+                subtensor.close()
+                bittensor.logging.debug("closing subtensor connection")
+
+    @staticmethod
+    def _run(cli: "bittensor.cli", subtensor: "bittensor.subtensor"):
+        console = Console()
```
Comment (basfroman) [inline bittensor/commands/stake.py:791]: should we use `bittensor.__console__` instead of `Console()` to be more consistent?

**Review by basfroman — DISMISSED** [review]
LGTM 
Left unblocking comments

**Review by thealligatorking — DISMISSED** [review]

File: bittensor/commands/stake.py (hunk 6 above)
Comment (opendansor) [reply bittensor/commands/stake.py:791]: For these commands, (that use this function) its mainly rich that gets used.

**Review by basfroman — APPROVED** [review]

**Pull request merged**